go/stellar/stellar.go

```diff
@@ -696,6 +696,130 @@ func sendPayment(mctx libkb.MetaContext, walletState *WalletState, sendArg SendP
 	}, nil
 }
 
+type SendPathPaymentArg struct {
+	From        stellar1.AccountID
+	To          stellarcommon.RecipientInput
+	Path        stellar1.PaymentPath
+	SecretNote  string
+	PublicMemo  string
+	QuickReturn bool
+}
+
+// SendPathPaymentCLI sends a path payment from CLI.
+func SendPathPaymentCLI(mctx libkb.MetaContext, walletState *WalletState, sendArg SendPathPaymentArg) (res SendPaymentResult, err error) {
+	return sendPathPayment(mctx, walletState, sendArg)
+}
+
+// SendPathPaymentGUI sends a path payment from GUI.
+func SendPathPaymentGUI(mctx libkb.MetaContext, walletState *WalletState, sendArg SendPathPaymentArg) (res SendPaymentResult, err error) {
+	return sendPathPayment(mctx, walletState, sendArg)
+}
+
+func sendPathPayment(mctx libkb.MetaContext, walletState *WalletState, sendArg SendPathPaymentArg) (res SendPaymentResult, err error) {
+	senderEntry, senderAccountBundle, err := LookupSender(mctx, sendArg.From)
+	if err != nil {
+		return res, err
+	}
+	senderSeed, err := stellarnet.NewSeedStr(senderAccountBundle.Signers[0].SecureNoLogString())
+	if err != nil {
+		return res, err
+	}
+	senderAccountID := senderEntry.AccountID
+
+	recipient, err := LookupRecipient(mctx, stellarcommon.RecipientInput(sendArg.To), false)
+	if err != nil {
+		return res, err
+	}
+	if recipient.AccountID == nil {
+		return res, errors.New("cannot send a path payment to a user without a stellar account")
+	}
+
+	baseFee := walletState.BaseFee(mctx)
+
+	to, err := stellarnet.NewAddressStr(recipient.AccountID.String())
+	if err != nil {
+		return res, err
+	}
+
+	sp, unlock := NewSeqnoProvider(mctx, walletState)
+	defer unlock()
+
+	sig, err := stellarnet.PathPaymentTransaction(senderSeed, to, sendArg.Path.SourceAsset, sendArg.Path.SourceAmountMax, sendArg.Path.DestinationAsset, sendArg.Path.DestinationAmount, AssetSliceToAssetBase(sendArg.Path.Path), sendArg.PublicMemo, sp, nil, baseFee)
+	if err != nil {
+		return res, err
+	}
```

Comment:
`sendPathPayment` looks on the whole the same as `sendPayment` except for this here `PathPaymentTransaction` and `walletState.SubmitPathPayment`. `sendPayment` already switches `PaymentXLMTransaction` by `if !funded {`. What about deduping `sendPathPayment` and `sendPayment` by taking an optional path-start?

timebounds are nil

Reply:
I also didn't want to mess with sendPayment.  It works.  This is new and an "advanced" feature that few users will use.

There are other differences too, like request clearing, in chat send notifications, etc.

I think it's fine for them to be split, especially now as a new feature.

I'd be more in favor of taking some of the common functionality out of both of them into little functions that they both call.

Reply:
It could be done in a way that would almost provably not change what sendPayment executes when `path == nil`. It may be hard to tell going forward the difference between purposeful differences and accidental desync. Which wouldn't be the case with `if path == nil`s. It'd be easy to split later but harder to merge. I approve whatever decision you make.